docs/extending/developing-code/extending/case-studies/Undefined/testing.rst

```diff
@@ -0,0 +1,62 @@
+Testing ``Undefined``
+=====================
+
+Now that we have a complete version, we are in a position to try the tests. Recall when we added the docstring we had:
+
+.. code-block::
+
+    >> ConditionalExpression[a, False]
+     = Undefined
+    >> Attributes[Undefined]
+     = {Protected}
+
+
+The lines that start with ``>>`` are examples to show and that can be tried. The lines that start with ``=`` are the expected output.
+See :ref:`Documentation Markup` for a full list and description of the homegrown markup.
+
+
+Understanding the markup, we can entry that in a mathics session to verify that we get expected results:
+
+
+.. code-block:: Bash
+
+
+    $ mathics
+
+    Mathics 5.0.3dev0
+    on CPython 3.8.12 (heads/v2.3.4.1_release:4a6b4d3504, Jun  3 2022, 15:46:12)
+    ...
+
+    In[1]:= ConditionalExpression[a, False]
+    Out[1]= Undefined
+
+    In[2]:= Attributes[Undefined]
+    Out[2]= {Protected}
+
+So far so good. However we can run the all of these example
```

Comment:
the all -> all

Reply:
Fixed now. Thanks for spotting - especially since this kind of thing is not caught by a spelling checker.